src/attributes/val.js

```diff
@@ -128,7 +129,12 @@ jQuery.extend( {
 			set: function( elem, value ) {
 				var optionSet, option,
 					options = elem.options,
-					values = jQuery.makeArray( value ),
+
+					// Strip and collapse whitespace to normalize set values
+					// https://html.spec.whatwg.org/#strip-and-collapse-whitespace
+					values = jQuery.map( jQuery.makeArray( value ), function( val ) {
+						return jQuery.trim( val ).replace( rspaces, " " );
+					} ),
```

Comment:
I'm confused... what is the point of this? Fixing gh-2978 requires an update to `valHooks.option.get`, not to `valHooks.select.set`: https://github.com/jquery/jquery/issues/2978#issuecomment-194447965 . Specifically, strip and collapse whitespace is necessary to get the value of an option when it has no `value` attribute, and AFAICT nowhere else.